python/ray/serve/tests/test_grpc.py

```diff
@@ -517,5 +518,71 @@ def Streaming(self, user_message):
     assert error_message in rpc_error.details()
 
 
+@pytest.mark.parametrize("streaming", [False, True])
+def test_grpc_proxy_cancellation(ray_instance, ray_shutdown, streaming: bool):
+    """Test gRPC request client cancelled.
+
+    When the request is canceled, gRPC proxy should cancel the underlying task.
+    """
+    grpc_port = 9000
+    grpc_servicer_functions = [
+        "ray.serve.generated.serve_pb2_grpc.add_UserDefinedServiceServicer_to_server",
+    ]
+
+    serve.start(
+        grpc_options=gRPCOptions(
+            port=grpc_port,
+            grpc_servicer_functions=grpc_servicer_functions,
+        ),
+    )
+
+    running_signal_actor = SignalActor.remote()
+    cancelled_signal_actor = SignalActor.remote()
+
+    @serve.deployment
+    class Downstream:
+        async def wait_for_singal(self):
+            await running_signal_actor.send.remote()
+            await send_signal_on_cancellation(cancelled_signal_actor)
+
+        async def __call__(self, *args):
+            await self.wait_for_singal()
+            return serve_pb2.UserDefinedResponse(greeting="hello")
+
+        async def Streaming(self, *args):
+            await self.wait_for_singal()
+            yield serve_pb2.UserDefinedResponse(greeting="hello")
+
+    @serve.deployment
+    class Ingress:
+        async def __call__(self, *args):
```

Comment:
Why does this need to run inside of a deployment? There are other tests doing this because they're explicitly testing the `DeploymentHandle` behavior within a deployment. In this case you're just using the gRPC client so no need.

Reply:
I mostly just copy the other test but changed it to send the gRPC requests to test the cancel logic, mostly just want to ensure this works for gRPC as well 😅 I can try to see if I can get this working outside of a deployment

Reply:
oh i see, I just need to mark this test `asyncio` then I can have those awaits XD